components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs

```diff
@@ -1600,14 +1602,16 @@ public sealed class GatewayServer : SessionServerBase, IServer {
                 ApplicationName = "Opc UA Gateway Server",
                 ApplicationType = Opc.Ua.ApplicationType.ClientAndServer,
                 ApplicationUri =
-                    $"urn:{Utils.GetHostName()}:Microsoft:OpcGatewayServer",
+                    $"urn:{Dns.GetHostName()}:Microsoft:OpcGatewayServer",
                 ProductUri = "http://opcfoundation.org/UA/SampleServer",
 
                 SecurityConfiguration = new SecurityConfiguration {
                     ApplicationCertificate = new CertificateIdentifier {
-                        StoreType = "Directory",
-                        StorePath =
-                "OPC Foundation/CertificateStores/MachineDefault",
+                        StoreType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                                "X509Store" : "Directory",
+                        StorePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                                "CurrentUser\\UA_MachineDefault" :
+                                "OPC Foundation/CertificateStores/MachineDefault",
```

Comment:
>OPC Foundation [](start = 33, length = 14)

prefix is rather Microsft, its our server

Reply:
did not want to touch this server in this step, but you're right, I'll change it